src/Dropdown/index.tsx

```diff
@@ -1,134 +1,119 @@
-import React, { useRef, useState, useEffect } from 'react'
+import React, { useRef, useState, useEffect, useLayoutEffect } from 'react'
 import styled, { css } from 'styled-components'
 
-interface IDropdownContentPosition {
+interface IPosition {
   left?: 0
   right?: 0
-  top?: string | 0
-  bottom?: string | 0
+  top?: number
+  bottom?: number
 }
 
-interface IDropdownContentProps {
-  position: IDropdownContentPosition
+interface IContentProps extends IPosition {
+  opened: boolean
 }
 
-const DropdownContent = styled.div<IDropdownContentProps>`
+const Content = styled.div<IContentProps>`
   position: absolute;
   background-color: #f9f9f9;
   min-width: 160px;
   box-shadow: 0 5px 15px rgba(0,0,0,0.08);
   padding: 20px;
   padding-right: 10%;
-  ${ ({ position }) => css`
-    ${typeof position.left !== 'undefined' && css`
-      left: ${position.left};
-    `}
-    ${typeof position.right !== 'undefined' && css`
-      right: ${position.right};
+  visibility: ${({ opened }) => opened ? 'visible' : 'hidden'};
+  ${ ({ left, right, top, bottom }) => css`
+    ${typeof left !== 'undefined' && css`
+      left: ${left};
     `}
-    ${typeof position.bottom !== 'undefined' && css`
-      bottom: ${position.bottom};
-      margin-bottom: 5px;
+    ${typeof right !== 'undefined' && css`
+      right: ${right};
     `}
-    ${typeof position.top !== 'undefined' && css`
-      top: ${position.top};
+    ${typeof top !== 'undefined' && css`
+      top: ${top}px;
       margin-top: 5px;
     `}
+    ${typeof bottom !== 'undefined' && css`
+      bottom: ${bottom}px;
+      margin-bottom: 5px;
+    `}
   `}
 `
 
-type Vertical = 'top' | 'bottom'
-type Horizontal = 'left' | 'right'
-type Layout = [Vertical, Horizontal] | 'auto'
+type Vertical = 'to-top' | 'to-bottom'
+type Horizontal = 'to-left' | 'to-right'
+interface ILayout {
+  horizontal: Horizontal
```

Comment:
I think I would prefer to have a separate horizontal and vertical prop - both with default values so you can just set a prop  with a primitive instead of an object (which could potentially cause unnecessary rerenders if its an object literal)